tests/stubtest_third_party.py

```diff
@@ -16,10 +19,23 @@
 from utils import PYTHON_VERSION, colored, get_mypy_req, print_error, print_success_msg
 
 
+@contextlib.contextmanager
+def temporarily_delete_stdlib_distutils(typeshed_dir: Path) -> Iterator[None]:
+    stdlib_distutils_dir = typeshed_dir / "stdlib" / "distutils"
+    with tempfile.TemporaryDirectory() as td:
+        shutil.copytree(stdlib_distutils_dir, td, dirs_exist_ok=True)
+        try:
+            shutil.rmtree(stdlib_distutils_dir)
+            yield
+        finally:
+            shutil.copytree(td, stdlib_distutils_dir, dirs_exist_ok=True)
```

Comment:
This should be simpler and faster:
```suggestion
        shutil.move(stdlib_distutils_dir, td)
        try:
            yield
        finally:
            shutil.move(td, stdlib_distutils_dir)
```

(As long as the tempdir and typeshed are on the same volume, this move() should just be a single operation to point one file to another place, while your copytree() solution needs to go over and copy every file.)

Reply:
Nice, TIL :)

Seems like we don't actually need any hacks anymore, though (no idea what changed since I first opened the PR) -- https://github.com/python/typeshed/pull/10948#issuecomment-2002027351